api/python/provisioner/commands/setup_provisioner.py

```diff
@@ -1803,6 +1803,18 @@ def _run(self, nodes, **kwargs):  # noqa: C901 FIXME
             ), targets=run_args.primary.minion_id
         )
 
+        # Grains data is not getting refreshed within sls files
+        # if we call init.sls for machine_id states.
+        logger.info("Refresh machine id on the system")
+        for state in [
+            'components.provisioner.config.machine_id.refresh_machine_id',
```

Comment:
@mazinamdar can't we just apply only `components.provisioner.config.machine_id`?

Reply:
I am facing same issue of grains refresh if I run `components.provisioner.config.machine_id`  so we need to run both states separately.

Reply:
Got it. Thanks